x-pack/plugins/aiops/public/components/change_point_detection/fields_config.tsx

```diff
@@ -138,25 +158,244 @@ export interface FieldPanelProps {
  * @constructor
  */
 const FieldPanel: FC<FieldPanelProps> = ({
+  panelIndex,
   fieldConfig,
   onChange,
   onRemove,
   removeDisabled,
   onSelectionChange,
   'data-test-subj': dataTestSubj,
 }) => {
-  const { combinedQuery, requestParams } = useChangePointDetectionContext();
+  const { embeddable } = useAiopsAppContext();
+
+  const { dataView } = useDataSource();
+
+  const { combinedQuery, requestParams, selectedChangePoints } = useChangePointDetectionContext();
 
   const splitFieldCardinality = useSplitFieldCardinality(fieldConfig.splitField, combinedQuery);
 
   const [isExpanded, setIsExpanded] = useState<boolean>(true);
 
+  const [isActionMenuOpen, setIsActionMenuOpen] = useState(false);
+
+  const [dashboardAttachment, setDashboardAttachment] = useState<{
+    applyTimeRange: boolean;
+    maxSeriesToPlot: number;
+  }>({
+    applyTimeRange: false,
+    maxSeriesToPlot: 6,
+  });
+  const [dashboardAttachmentReady, setDashboardAttachmentReady] = useState<boolean>(false);
+
   const {
     results: annotations,
     isLoading: annotationsLoading,
     progress,
   } = useChangePointResults(fieldConfig, requestParams, combinedQuery, splitFieldCardinality);
 
+  const openCasesModalCallback = useCasesModal(EMBEDDABLE_CHANGE_POINT_CHART_TYPE);
+
+  const selectedPartitions = useMemo(() => {
+    return (selectedChangePoints[panelIndex] ?? []).map((v) => v.group?.value as string);
+  }, [selectedChangePoints, panelIndex]);
+
+  const caseAttachmentButtonDisabled =
+    isDefined(fieldConfig.splitField) && selectedPartitions.length === 0;
+
+  const panels: EuiContextMenuProps['panels'] = [
```

Comment:
Looks like you need to add checks that the user has write access to dashboard and cases. Here I have a user with `read` permissions for Dashboard and Cases, but the context menu is still showing these actions:

<img width="370" alt="image" src="https://github.com/elastic/kibana/assets/7405507/90764f41-3633-43b4-bfd5-b7cb869c6ce2">

Reply:
I was able to reproduce this as well

Reply:
Capabilities check was added in https://github.com/elastic/kibana/pull/162796/commits/5d0bbf30e4f669227115c0d4beba05d1b76024fd

Reply:
Capabilities checks look to be working as expected now in my testing.